### PR TITLE
DataViews: Type the BulkActions component

### PR DESCRIPTION
**Pull request by youknowriad**
Related #55083 
Follow-up to #61185

## What?

The DataViews package is a types heavy package. There's a lot of structures that need to be documented properly. So this continues the effort to add explicit types to the package. This PR focuses on typing the BulkActions component.

## Testing instructions

There's no functional change, it's essentially a code quality + documentation change.

### CODE REVIEW

File: packages/dataviews/src/bulk-actions.tsx [hunk 1]
```diff
@@ -21,19 +22,55 @@ const {
...
+	getItemId: ( item: Item ) => string;
+}
+
+export function useHasAPossibleBulkAction( actions: Action[], item: Item ) {
 	return useMemo( () => {
 		return actions.some( ( action ) => {
-			return action.supportsBulk && action.isEligible( item );
+			return (
+				action.supportsBulk &&
+				( ! action.isEligible || action.isEligible( item ) )
```
Comment (youknowriad) [inline packages/dataviews/src/bulk-actions.tsx:57]: isEligible is not mandatory in actions (you can check that in other components). So typing this component actually highlights that this case was not being considered. So this should be a fix.

File: packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ export const deleteAction = {
 	},
 	hideModalHeader: true,
 	supportsBulk: true,
-	RenderModal: ( { items, closeModal, onPerform } ) => {
+	RenderModal: ( { items, closeModal, onActionPerformed } ) => {
```
Comment (youknowriad) [inline packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js:107]: If I'm not wrong, this was a bug too. I see that all actions everywhere in dataviews use `onActionPerformed` prop and not `onPerform`. This is also something that I found thanks to the types.

File: packages/dataviews/src/bulk-actions.tsx (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/dataviews/src/bulk-actions.tsx:57]: Good catch 👍

File: packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js (hunk 2 above)
Comment (jorgefilipecosta) [reply packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js:107]: Yes, it seems it was a bug.

**Review by jorgefilipecosta — COMMENTED** [review]
It looks good and the addition of types caught two issues 👍

**Review by jorgefilipecosta — COMMENTED** [review]
It looks good and the addition of types caught two issues 👍

**Review by jorgefilipecosta — APPROVED** [review]

**Pull request merged**